packages/contract-call-decoder/src/lib/ContractCallDecoder.ts

```diff
@@ -0,0 +1,210 @@
+import radspec from '@blossom-labs/rosette-radspec';
+import { Transaction as TransactionRosette } from '@blossom-labs/rosette-radspec/dist/declarations/src/types/web3';
+import { decode as decodeBytecode } from '@ethereum-sourcify/bytecode-utils';
+import {
+  Fragment,
+  FunctionFragment,
+  Interface,
+  JsonFragment,
+} from '@ethersproject/abi';
+import { Provider } from '@ethersproject/providers';
+import { EthereumProvider } from 'ethereum-provider';
+
+import { extractCustomFields, getValueFromDecodedFunctionData } from './utils';
+
+type Transaction = TransactionRosette & {
+  readonly chainId?: number;
+};
+
+require('isomorphic-fetch');
+
+export enum MetadataSources {
+  Sourcify,
+  BytecodeMetadata,
+}
+
+type GetMetadataOptions = {
+  readonly source?: MetadataSources;
+  readonly chainId?: number;
+  readonly address?: string;
+  readonly rpcProvider?: EthereumProvider;
+  readonly ipfsGateway?: string;
+  readonly sourcifyProvider?: string;
+};
+
+const defaultGetMetadataOptions: GetMetadataOptions = {
+  source: MetadataSources.Sourcify,
+  sourcifyProvider: 'https://repo.sourcify.dev',
+  ipfsGateway: 'https://cloudflare-ipfs.com/',
+};
+
+export async function getMetadataFromAddress(options: GetMetadataOptions) {
+  options = { ...defaultGetMetadataOptions, ...options };
+  // eslint-disable-next-line functional/no-let
+  let contractMetadataJSON;
+  if (options.source === MetadataSources.Sourcify) {
+    try {
+      const req = await fetch(
+        `${options.sourcifyProvider}/contracts/full_match/${options.chainId}/${options.address}/metadata.json`
+      );
+      contractMetadataJSON = await req.json();
+    } catch (e) {
+      console.log(e);
+      return false;
```

Comment:
I was just following the functional way to do so, but it makes no sense since the rest of the project doesn't.
https://github.com/eslint-functional/eslint-plugin-functional/blob/main/docs/rules/no-throw-statement.md